src/qibo/tomography/gate_set_tomography.py

```diff
@@ -286,12 +292,25 @@ def GST(
 
     for gate in gate_set:
         if gate is not None:
-            nqubits = len(gate.qubits)
-            if nqubits not in (1, 2):
+            init_args = signature(gate[0]).parameters
+            params = gate[1]
+
+            angle_names = [name for name in init_args if name in {"theta", "phi"}]
+
+            angle_values = {}
+            for name, value in zip(angle_names, params):  # Zip ensures correct order
+                angle_values[name] = value
```

Comment:
Thanks Andrea! I realise that if we do
``` python
for angle, param in zip(["theta", "phi"][:len(gate[1])], gate[1])
```
it won't work for gates like `gates.GPI(0, np.pi)` because `GPI` gate doesn't have `theta` parameter. I run into this error message:
```
---> 46 gate = gate(*range(nqubits), **params)

TypeError: GPI.__init__() got an unexpected keyword argument 'theta'
```
Hence my train of thought is that if we search for `theta` or `phi` or both, then we just extract that angle_name out and assign the parameter directly (Lines 295 to 302)
``` python
angle_names = [name for name in init_args if name in {"theta", "phi"}]

angle_values = {}
for name, value in zip(angle_names, params):  # Zip ensures correct order
    angle_values[name] = value
```